### ISSUE TITLE
Bug in CosineAnnealingLR in Python 2 (?)

### ISSUE THREAD

**Issue by nguyen-binh-minh** [pytorch/pytorch#4648]
The learning rate only takes two values `base_lr` or `eta_min` if the `T_max` parameter passed in the constructor is an integer. This is probably because `self.last_epoch / self.T_max` evaluates to an integer in Python 2.

**Comment by apaszke** [comment]
Ouch, can you please send a PR?
